pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -839,6 +825,36 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 						Service:                    service,
 					}
 
+					// Hack for Kubernetes stateful sets/headless services
+					// Instead of generating a single 0.0.0.0:Port listener, generate N listeners
+					// (where N <= 5 at most) one for each podIP.
+					if bind == "" && service.Resolution == model.Passthrough &&
+						service.Attributes.ServiceRegistry == string(serviceregistry.KubernetesRegistry) {
+						instances, err := env.InstancesByPort(service, servicePort.Port, nil)
```

Comment:
This method is already insanely long ... can turn this code block into a helper method?  Possibly move this to `configgen.buildSidecarOutboundListenerForPortOrUDS` or a method along side it?

Reply:
Its not about the length. Rather, its about where the logic fits . It won't fit in the buildSidecarOutboundListenerFor.. function. Creating another helper will simply cause more scrolling as you bounce back and forth between functions to understand the core logic. 
Btw, the length is due to the long comments in this function and the way the structs are laid out, one field per line. Not due to the amount of code in the function.